lib/messages/blob.js

```diff
@@ -19,32 +20,51 @@ class Blob extends Message {
   }
 
   getCore() {
-    let lines = Buffer.from(this.blob.content, 'base64').toString().split(/\n/);
+    const buffer = Buffer.from(this.blob.content, 'base64');
     let title = this.blob.path;
     let htmlURL = this.blob.html_url;
 
-    if (this.start) {
-      lines = lines.slice(
-        Number(this.start) - 1,
-        Number(this.end || this.start),
-      );
+    // only unfurl binary files that are images
+    if (isBinaryFile.sync(buffer, this.blob.size)) {
+      const imageExtensions = ['png', 'jpg', 'jpeg', 'gif'];
+      const blobExtension = this.blob.name.toLowerCase().split('.')[1];
```

Comment:
I wasn't too sure on how to check that the file is an image (since the contents API doesn't go into detail on _what_ the file is), so I just made a whitelist of acceptable image extensions and grabbed the file extension from the blob.